osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SecretManagementService.java

```diff
@@ -0,0 +1,165 @@
+
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.lang3.NotImplementedException;
+import org.opensmartgridplatform.adapter.protocol.dlms.application.wsclient.SecretManagementClient;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.SecurityKeyType;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+import org.opensmartgridplatform.shared.exceptionhandling.EncrypterException;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalException;
+import org.opensmartgridplatform.shared.security.EncryptedSecret;
+import org.opensmartgridplatform.shared.security.EncryptionProviderType;
+import org.opensmartgridplatform.shared.security.Secret;
+import org.opensmartgridplatform.shared.security.providers.RsaEncryptionProvider;
+import org.opensmartgridplatform.ws.schema.core.secret.management.GetSecretsRequest;
+import org.opensmartgridplatform.ws.schema.core.secret.management.GetSecretsResponse;
+import org.opensmartgridplatform.ws.schema.core.secret.management.SecretType;
+import org.opensmartgridplatform.ws.schema.core.secret.management.SecretTypes;
+import org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecret;
+import org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecrets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class SecretManagementService implements SecurityKeyService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SecretManagementService.class);
+    private final RsaEncryptionProvider rsaEncryptionProvider;
+    private final SecretManagementClient secretManagementClient;
+
+    public SecretManagementService(RsaEncryptionProvider rsaEncryptionProvider,
+            SecretManagementClient secretManagementClient) {
+        this.rsaEncryptionProvider = rsaEncryptionProvider;
+        this.secretManagementClient = secretManagementClient;
+    }
+
+    @Override
+    public byte[] reEncryptKey(byte[] externallyEncryptedKey, SecurityKeyType keyType) throws FunctionalException {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] decryptKey(byte[] encryptedKey, SecurityKeyType keyType) throws ProtocolAdapterException {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] encryptKey(byte[] plainKey, SecurityKeyType keyType) throws ProtocolAdapterException {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] getDlmsMasterKey(String deviceIdentification) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] getDlmsAuthenticationKey(String deviceIdentification) throws EncrypterException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.opensmartgridplatform.shared.exceptionhandling.EncrypterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)